driver-reactive-streams/src/main/com/mongodb/reactivestreams/client/internal/MongoOperationPublisher.java

```diff
@@ -445,23 +445,24 @@ public boolean getRetryReads() {
                                         clientSession);
     }
 
-    <R> Mono<R> createReadOperationMono(
+    <R> Publisher<R> createReadOperationMono(
             final Supplier<AsyncReadOperation<R>> operation,
             @Nullable final ClientSession clientSession) {
         return createReadOperationMono(operation, clientSession, getReadPreference());
     }
 
-    <R> Mono<R> createReadOperationMono(
+    <R> Publisher<R> createReadOperationMono(
             final Supplier<AsyncReadOperation<R>> operation,
             @Nullable final ClientSession clientSession,
             final ReadPreference readPreference) {
         AsyncReadOperation<R> readOperation = operation.get();
         return executor.execute(readOperation, readPreference, getReadConcern(), clientSession);
     }
 
-    <R> Mono<R> createWriteOperationMono(final Supplier<AsyncWriteOperation<R>> operation, @Nullable final ClientSession clientSession) {
-        AsyncWriteOperation<R> writeOperation = operation.get();
-        return executor.execute(writeOperation, getReadConcern(), clientSession);
+    <R> Mono<R> createWriteOperationMono(final Supplier<AsyncWriteOperation<R>> operation,
+            @Nullable final ClientSession clientSession) {
+        // TODO: is this kosher to wrap a custom Publisher which wraps a Mono
+        return Mono.from(executor.execute(operation.get(), getReadConcern(), clientSession));
```

Comment:
from what I understand, this is just to avoid exposing `Mono` type in the signature, but for all cases where `Context` would be relevant, we'd get a `Mono` anyway?

Reply:
This has been changed now. The execute method now returns a Mono.  OK to resolve?

Reply:
sounds good